xarray/core/dataarray.py

```diff
@@ -6670,6 +6674,72 @@ def resample(
             **indexer_kwargs,
         )
 
+    def to_dask_dataframe(
+        self,
+        dim_order: Sequence[Hashable] | None = None,
+        set_index: bool = False,
+        *,
+        name: Hashable | None = None,
+    ) -> DaskDataFrame:
+        """Convert this array into a dask.dataframe.DataFrame.
+
+        Parameters
+        ----------
+        dim_order : Sequence of Hashable or None , optional
+            Hierarchical dimension order for the resulting dataframe.
+            Array content is transposed to this order and then written out as flat
+            vectors in contiguous order , so the last dimension in this list
+            will be contiguous in the resulting DataFrame. This has a major influence
+            on which operations are efficient on the resulting dask dataframe.
+
+        name : Hashable or None, optional
+            Name given to this array(required if unnamed).
+            It is a keyword-only argument. A keyword-only argument can only be passed
+            to the function using its name as a keyword argument , and not as a
+            positional argument.
+
+        set_index : bool, default: False
+            If set_index=True, the dask DataFrame is indexed by this dataset's
+            coordinate. Since dask DataFrames do not support multi-indexes,
+            set_index only works if the dataset only contains one dimension.
+
+        Returns
+        -------
+        dask.dataframe.DataFrame
+
+        Examples
+        --------
+        >>> da = xr.DataArray(
+        ...     np.arange(4 * 3 * 2).reshape(4, 3, 2),
+        ...     dims=("time", "lat", "lon"),
+        ...     coords={
+        ...         "time": np.arange(4),
+        ...         "lat": [-30, -20, -10],
+        ...         "lon": [120, 130],
+        ...     },
+        ...     name="temperature",
+        ...     attrs={"units": "Celsius", "description": "Random temperature data"},
+        ... )
+        >>> da.to_dask_dataframe(["lat", "lon", "time"], name="temp_dataframe")
+        Dask DataFrame Structure:
+                        lat    lon   time temp_dataframe
+        npartitions=1
+        0              int64  int64  int64          int64
+        23               ...    ...    ...            ...
+        Dask Name: concat-indexed, 1 graph layer
+
+        """
+        if name is None:
+            name = self.name
+
+        if name is None:
+            raise ValueError(
+                "Cannot convert an unnamed DataArray to a "
+                "dask dataframe : use the ``name`` parameter ."
+            )
+        ds = self._to_dataset_whole(name)
+        return ds.to_dask_dataframe(dim_order, set_index)
```

Comment:
```suggestion
        name = self.name if self.name is not None else _THIS_ARRAY
        ds = self._to_dataset_whole(name, shallow_copy=False)
        return ds.to_dask_dataframe(dim_order, set_index)
```

I think we go with this. I don't think it should be necessary to name the dataarray which is more in line with how `self._to_temp_dataset` works and setting `dataarray.name = "new_name"` is easy enough.